megamek/src/megamek/common/weapons/WeaponHandler.java

```diff
@@ -1173,33 +1173,34 @@ public boolean handle(IGame.Phase phase, Vector<Report> returnedReports) {
                         r.subject = subjectId;
                         vPhaseReport.addElement(r);
                         hits = 0;
-                    }
-                    // targeting a hex for igniting
-                    if ((target.getTargetType() == Targetable.TYPE_HEX_IGNITE)
+                    // targeting a hex for igniting    
+                    } else if ((target.getTargetType() == Targetable.TYPE_HEX_IGNITE)
                             || (target.getTargetType() == Targetable.TYPE_BLDG_IGNITE)) {
                         handleIgnitionDamage(vPhaseReport, bldg, hits);
                         hits = 0;
-                    }
                     // targeting a hex for clearing
-                    if (target.getTargetType() == Targetable.TYPE_HEX_CLEAR) {
+                    } else if (target.getTargetType() == Targetable.TYPE_HEX_CLEAR) {
                         nDamage = nDamPerHit * hits;
                         handleClearDamage(vPhaseReport, bldg, nDamage);
                         hits = 0;
-                    }
                     // Targeting a building.
-                    if (target.getTargetType() == Targetable.TYPE_BUILDING) {
+                    } else if (target.getTargetType() == Targetable.TYPE_BUILDING) {
                         // The building takes the full brunt of the attack.
                         nDamage = nDamPerHit * hits;
                         handleBuildingDamage(vPhaseReport, bldg, nDamage,
                                 target.getPosition());
                         hits = 0;
-                    }
-                    if (entityTarget != null) {
+                    } else if (entityTarget != null) {
                         handleEntityDamage(entityTarget, vPhaseReport, bldg,
                                 hits, nCluster, bldgAbsorbs);
                         server.creditKill(entityTarget, ae);
                         hits -= nCluster;
                         firstHit = false;
+                    } else {
+                        // we shouldn't be here, but if we get here, let's set hits to 0
+                        // to avoid infinite loops
+                        hits = 0;
+                        getLogger().error(getClass(), "Unexpected target type: " + target.getTargetType());
```

Comment:
```suggestion
                        getLogger().error(this, "Unexpected target type: " + target.getTargetType());
```

Reply:
Under the hood, the 'error' method just calls 'getClass()' anyway, so I don't see why this is necessary.